src/main/java/com/hbm/tileentity/machine/TileEntityFluidPump.java

```diff
@@ -0,0 +1,223 @@
+package com.hbm.tileentity.machine;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.hbm.interfaces.IFluidAcceptor;
+import com.hbm.interfaces.IFluidSource;
+import com.hbm.inventory.fluid.FluidType;
+import com.hbm.inventory.fluid.Fluids;
+import com.hbm.inventory.fluid.tank.FluidTank;
+import com.hbm.lib.Library;
+import com.hbm.tileentity.TileEntityMachineBase;
+
+import api.hbm.energy.IEnergyUser;
+import api.hbm.fluid.IFluidStandardTransceiver;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraftforge.common.util.ForgeDirection;
+import net.minecraft.block.Block;
+import net.minecraft.init.Blocks;
+import net.minecraft.world.World;
+
+public class TileEntityFluidPump extends TileEntityMachineBase implements IFluidAcceptor, IFluidSource, IEnergyUser, IFluidStandardTransceiver {
```

Comment:
`IFluidSource` is part of the deprecated old fluid system and shouldn't be implemented in all newer TEs.

`IFluidStandardTransceiver` is for TEs that both have in and output tanks, but the pump only outputs so it should be `IFluidStandardSender` instead.

`TileEntityMachineBase` is for TEs with slots, you should extend `TileEntityLoadedBase` and implement `INBTPacketReceiver` for packet sync.